### ISSUE TITLE
Windows + MSVC issues

### ISSUE THREAD

**Issue by piponazo** [Exiv2/exiv2#487]
I noticed few things when trying to debug on Windows + Visual Studio 2017. 

1. I can debug the **exiv2** application but I cannot access the exiv2 library symbols when I try to step into a exiv2 function/class. On linux I can debug with gcc/clang without issues.
2. The compilation of the **exiv2** application with exiv2lib as a STATIC library fails (`-DBUILD_SHARED_LIBS=OFF`).

**TODO**: Investigate more about these issues and complete the information in this description.

**Comment by clanmills** [comment]
I'll investigate this today because it's useful for both the localtime.c and TZ issue.  I'll update you later.

**Comment by piponazo** [comment]
Regarding to the debugging issue, it seems that we are not generating a PDB file for the **exiv2lib** target:

```
pipon@DesktopWin10 MINGW64 /f/projects/exiv2/buildDebugMSVC (conanWebready)
$ find -name *.pdb
./bin/exiv2.pdb
./lib/Debug/xmp.pdb
./src/exiv2.dir/Debug/vc141.pdb
./src/exiv2lib.dir/Debug/vc141.pdb
./src/exiv2lib_int.dir/Debug/exiv2lib_int.pdb
./xmpsdk/xmp.dir/Debug/xmp.pdb
```
I'll check the compiler flags on MSVC and try to determine what's happening there.

**Comment by clanmills** [comment]
The issue with `-DBUILD_SHARED_LIBS=OFF` is being caused by Conan/Expat requiring the C Runtime library /MTd.  I've successfully built this and it executes.  However, **I'm unable to step/debug in Visual Studio** because it hasn't generated the .pdb files.  And I haven't yet discovered why not, it looks OK in the IDE.

Here's how to build it:

I've used the following profile `msvc2017Debug64Static`:

```
[build_requires]
[settings]
arch=x86_64
build_type=Debug
compiler=Visual Studio
compiler.runtime=MTd
compiler.version=15
os=Windows
arch_build=x86_64
os_build=Windows
[options]
[env]
```

And I've built with:
```
> conan install .. --profile msvc2017Debug64Static --build missing
> cmake         .. -G "Visual Studio 15 2017 Win64" -DCMAKE_BUILD_TYPE=Debug -DBUILD_SHARED_LIBS=Off
> cmake --build . --config Debug
```

If you fail to use `compiler.runtime=MTd`, you'll build a library:

```
C:\Users\rmills\gnu\github\exiv2\exiv2\build>dir/s *.lib
 Directory of C:\Users\rmills\gnu\github\exiv2\exiv2\build\lib\Debug
2018-10-16  01:59 PM        71,580,270 exiv2lib.lib
2018-10-16  01:58 PM        10,163,530 xmp.lib
 Directory of C:\Users\rmills\gnu\github\exiv2\exiv2\build\src\exiv2lib_int.dir\Debug
2018-10-16  01:58 PM        24,680,928 exiv2lib_int.lib
C:\Users\rmills\gnu\github\exiv2\exiv2\build>
```
However he fails to link any application (eg xmpsample), because he's looking for `__imp__mbspbrk` required by `_Expat_LoadLibrary()` which is in the Static C run-time library (/MTd)

```
"C:\Users\rmills\gnu\github\exiv2\exiv2\build\samples\xmpsample.vcxproj" (default target) (42) ->
  expat.lib(loadlibrary.obj) : error LNK2019: unresolved external symbol __imp__mbspbrk referenced in function _Expat_LoadLibrary [C:\Users\rmills\gnu\github\exiv2\exiv2\build\s
amples\xmpsample.vcxproj]
```
With the v0.26 msvc/ hand-built sln/vcproj files, I took the decision "all DLLs" or "all Static".  It looks as though conan has a similar philosophy.

```
C:\Users\rmills\gnu\github\exiv2\exiv2\build>dir bin\exiv2.exe
 Directory of C:\Users\rmills\gnu\github\exiv2\exiv2\build\bin
2018-10-16  02:20 PM         8,227,840 exiv2.exe

C:\Users\rmills\gnu\github\exiv2\exiv2\build>bin\exiv2 -vV --grep library
exiv2 0.27.0.1
library=C:\WINDOWS\SYSTEM32\ntdll.dll
library=C:\WINDOWS\System32\KERNEL32.DLL
library=C:\WINDOWS\System32\KERNELBASE.dll
library=C:\WINDOWS\System32\SHELL32.dll
library=C:\WINDOWS\System32\msvcrt.dll
library=C:\WINDOWS\System32\cfgmgr32.dll
library=C:\WINDOWS\System32\ucrtbase.dll
library=C:\WINDOWS\System32\shcore.dll
library=C:\WINDOWS\System32\RPCRT4.dll
library=C:\WINDOWS\System32\combase.dll
library=C:\WINDOWS\System32\bcryptPrimitives.dll
library=C:\WINDOWS\System32\windows.storage.dll
library=C:\WINDOWS\System32\advapi32.dll
library=C:\WINDOWS\System32\sechost.dll
library=C:\WINDOWS\System32\shlwapi.dll
library=C:\WINDOWS\System32\GDI32.dll
library=C:\WINDOWS\System32\gdi32full.dll
library=C:\WINDOWS\System32\msvcp_win.dll
library=C:\WINDOWS\System32\USER32.dll
library=C:\WINDOWS\System32\win32u.dll
library=C:\WINDOWS\System32\kernel.appcore.dll
library=C:\WINDOWS\System32\powrprof.dll
library=C:\WINDOWS\System32\profapi.dll
library=C:\WINDOWS\System32\PSAPI.DLL
library=C:\WINDOWS\System32\PSAPI.DLL
library=C:\WINDOWS\System32\PSAPI.DLL

C:\Users\rmills\gnu\github\exiv2\exiv2\build>
```

**Comment by piponazo** [comment]
In #493 I propose a fix to this issue, by changing a bit our CMake configuration.

However, what you said it's true, the culprit of this situation is the Expat dependency which is just a dependency for building the XMP code (not really something needed by the exiv2 library). 

I have taken a look to the expat new release (2.2.6) and it seems they have improved their CMake code and know you can control which C Runtime Library to use (Static or Dynamic). By moving to that version of expat we could even simplify more our CMake configuration. I'll create an issue about this, but this is something we can leave for the future (not critical for 0.27).

**Comment by clanmills** [comment]
I use Expat in samples/geotag.cpp.  You're right, expat is only used by xmp in the library.

**Comment by clanmills** [comment]
Yes.  That's a great find about expat 2.2.6.  Much simpler.  Yes, let's use /MD (/MDd) for both static and dynamic builds.

There was an issue earlier this year which I solved it with expat 2.2.5.  https://github.com/Exiv2/exiv2/issues/371

So, yes - let's request 2.2.6 with Conan.  The next time I update README.md, I'll document this.
